_observing-your-data/query-insights/query-insights-dashboard.md

```diff
@@ -78,22 +78,40 @@ The **Refresh** button reloads the query data based on the selected filters and
-| **Timestamp**           | When the query was run.                                                     |
-| **Latency**             | The time taken for the query to execute.                                    |
-| **CPU Time**            | The CPU resources consumed by the query.                                    |
-| **Memory Usage**        | The memory usage of the query.                                              |
-| **Indexes**             | The index or indexes on which the query was executed.                       |
-| **Search Type**         | The type of search used, for example, `query then fetch`.                   |
-| **Coordinator Node ID** | The node that coordinated the query.                                        |
-| **Total Shards**        | The total number of shards involved in running the query.                    |
+The metrics table dynamically adapts based on your **Type** filter selection (**Query**, **Group**, or both). Dynamic columns improve clarity by showing only the relevant data for each query type. The interface adapts in the following ways:
 
+![Column Display for Query Selected](../../images/Query-Insights/OnlyQueryColDisplay.png)
+
+When you select **queries only**, the table displays individual metrics, including **Latency**, **CPU Time**, and **Memory Usage**. The **Query Count** column isn't displayed because each row represents a single query.
+
+![Column Display for Group Selected](../../images/Query-Insights/OnlyGroupColDisplay.png)
+
+When you select **groups only**, the table displays aggregated metrics, including **Average Latency**, **Average CPU Time**, and **Average Memory Usage**. The **Query Count** column shows how many queries are in each group.
+
+![Column Display for Both Selected](../../images/Query-Insights/BothColDisplay.png)
+
+When you select both **groups** and **queries**, the table displays combined metrics, including both averaged and raw values, such as **Avg Latency/Latency**. 
+
+The following table provides descriptions for each metric and the metric's related query and group when selected.
+
+| Column name | Description  | Query selected | Group selected | Query + group selected |
+| :--- | :--- | :--- | :--- | :--- |
+| **ID**                  | Unique identifier for the query or group. | `ID`   | `ID`   | `ID`  |
+| **Type**                | Indicates whether the entry is a query or a group. | `Type`  | `Type` | `Type`  |
+| **Query Count**         | The number of queries aggregated in the group.  | Not shown  | `Query Count`        | `Query Count` (only for group)   |
+| **Timestamp**           | The amount of time when the query or group was recorded (may be empty for groups). | `Timestamp`     | Not shown            | `Timestamp` (only for query)     |
+| **Latency**             | The amount of time taken for individual queries to execute.  | `Latency`          | `Average Latency`    | `Avg Latency/Latency`          |
```

Comment:
Line 102: In parens: "only for the query"?

Reply:
Looking at this again, I think we can remove the parenthesis completely. The definitions already make it clear which element the metric pertains to.